crates/adapters/src/format/json/output.rs

```diff
@@ -109,9 +109,16 @@ struct JsonEncoder {
 }
 
 impl JsonEncoder {
-    fn new(output_consumer: Box<dyn OutputConsumer>, config: JsonEncoderConfig) -> Self {
+    fn new(output_consumer: Box<dyn OutputConsumer>, mut config: JsonEncoderConfig) -> Self {
```

Comment:
it is unusual for constructors to modify their arguments

Reply:
The argument is passed by value, so there is not side effect here.